### PR TITLE
Adding provisioner example for status code check

### PR DESCRIPTION
**Pull request by bendbennett**
For versions of Terraform prior to `v1.2.0` [Precondition and Postcondition](https://www.terraform.io/language/expressions/custom-conditions) checks are not available.

In the absence of precondition and postconditon checks a provisioner can be used to verify whether the response status code matches a desired value. Provisioner failure can then be used to determine whether to halt Terraform apply.

### CODE REVIEW

File: docs/data-sources/http.md [hunk 1]
```diff
@@ -92,6 +92,28 @@ resource "random_uuid" "example" {
...
+
+  # Optional request headers
+  request_headers = {
+    Accept = "application/json"
+  }
+}
+
+resource "null_resource" "example" {
+  provisioner "local-exec" {
+    command = contains([201, 204], data.http.example.status_code)
```
[the remark this answers is not in the document]
Comment (bendbennett) [reply docs/data-sources/http.md:116]: Have added code comment.

[the remark this answers is not in the document]
Comment (bflad) [reply docs/data-sources/http.md:116]: Looks good!

**Review by bflad — APPROVED** [review]
I believe we're covering this in #161 as well, but looks good to me -- we can sort out Windows-based shell issues as a followup, if necessary.

**Comment by bendbennett** [comment]
@bflad apologies. I meant to close this out as it is superseded by #160

**Pull request closed without merging**